### PR TITLE
fix(dependency): loosen the filename check for clang in CC/CXX

### PR DESCRIPTION
**Pull request by jochil**
... to cover cases like `clang++-13' or with prefixes.

### CODE REVIEW

File: pkg/dependencies/version.go [hunk 1]
```diff
@@ -46,7 +47,7 @@ func clangVersion(dep *Dependency, clangCheck execCheck) (*semver.Version, error
 	checkPath := false
 
 	if cc, found := os.LookupEnv("CC"); found {
-		if path.Base(cc) == "clang" {
+		if strings.Contains(path.Base(cc), "clang") {
```
Comment (fmeum) [inline pkg/dependencies/version.go:50]: I think we could start with `strings.HasPrefix`, which should be loose enough.

File: pkg/dependencies/version.go [hunk 2]
```diff
@@ -61,7 +62,7 @@ func clangVersion(dep *Dependency, clangCheck execCheck) (*semver.Version, error
 	}
 
 	if cxx, found := os.LookupEnv("CXX"); found {
-		if path.Base(cxx) == "clang++" {
+		if strings.Contains(path.Base(cxx), "clang") {
```
Comment (fmeum) [inline pkg/dependencies/version.go:65]: Should we still match `clang++` here?

**Review by fmeum — APPROVED** [review]

**Comment by fmeum** [comment]
I added https://github.com/CodeIntelligenceTesting/cifuzz/pull/315/commits/164685e05e8d2c515e47b43ad2caa322d69d89df to my PR

File: pkg/dependencies/version.go (hunk 1 above)
Comment (jochil) [reply pkg/dependencies/version.go:50]: It is the second time I adjust this check, so I wanted to make sure it is the last time :see_no_evil: 
I can easily image people having self compiled clang version with prefixes

File: pkg/dependencies/version.go (hunk 2 above)
Comment (jochil) [reply pkg/dependencies/version.go:65]: see above... the check was too strict, again. Now I would prefer to have it a little bit more tolerant :)

**Pull request merged**